File: lib/models/interaction_result.dart
class InteractionResult {
  final String drugA;
  final String drugB;
  final String severity;
  final String description;
  final String management;

  const InteractionResult({
    required this.drugA,
    required this.drugB,
    required this.severity,
    required this.description,
    required this.management,
  });

  factory InteractionResult.fromJson(Map<String, dynamic> json) {
    return InteractionResult(
      drugA: json['drug_a'] as String,
      drugB: json['drug_b'] as String,
      severity: json['severity'] as String,
      description: json['description'] as String,
      management: json['management'] as String,
    );
  }
}

File: PillChecker/PillChecker/Models/InteractionResult.swift
import Foundation

struct InteractionResult: Codable, Identifiable, Hashable {
    let drugA: String
    let drugB: String
    let severity: String
    let description: String
    let management: String

    var id: String { "\(drugA)-\(drugB)" }

    enum CodingKeys: String, CodingKey {
        case drugA = "drug_a"
        case drugB = "drug_b"
        case severity, description, management
    }
}
